python/pfs/drp/qa/detectorMapQa.py

```diff
@@ -0,0 +1,403 @@
+                    flist[aa].append(f)
+            self.log.info("Fiber number ({}{}): {}".format(arm[aa], spectrograph, len(flist[aa])))
+            self.log.info("Measured line ({}{}): {}".format(arm[aa], spectrograph, np.sum(measured[aa])))
+
+        plt.figure()
+        fcommon = set(flist[0]) | set(flist[1])
+        fibers = {}
+        difference = {}
+        wcommon = []
+        goodLines = [
+            "630.20",
+            "636.55",
+            "937.69",
+            "937.85",
+            "942.23",
+            "947.94",
+            "952.20",
+            "957.00",
+            "970.19",
+            "972.25",
+            "979.21",
+            "979.38",
+            "980.24",
+        ]
+        for f in fcommon:
+            b0 = (arcLines[0].fiberId == f) & measured[0]
+            b1 = (arcLines[1].fiberId == f) & measured[1]
+            wav0 = set(arcLines[0][b0].wavelength)
+            wav1 = set(arcLines[1][b1].wavelength)
+            wav = list(wav0 & wav1)
+            if len(wav) > 0:
+                wav.sort()
+                for w in wav:
+                    if "{:.2f}".format(w) in goodLines:
+                        if w not in wcommon:
+                            wcommon.append(w)
+                            fibers[w] = []
+                            difference[w] = []
+                        y = [aa[(aa.fiberId == f) & (aa.wavelength == w)].y[0] for aa in arcLines]
+                        fibers[w].append(f)
+                        difference[w].append(
+                            detectorMap[0].findWavelength(fiberId=f, row=y[0]) - detectorMap[1].findWavelength(
+                                fiberId=f, row=y[1])
+                        )
+        plt.figure()
+        wcommon.sort()
+        for w in wcommon:
+            self.log.info(
+                "{} nm ({} fibers, median={:.1e} nm, 1sigma={:.3f} nm)".format(
+                    w, len(fibers[w]),
+                    np.median(difference[w]),
+                    iqr(difference[w]) / 1.349)
+            )
+            plt.scatter(fibers[w], difference[w], s=3,
+                        label="{} nm ({} fibers, median={:.1e} nm, 1sigma={:.3f} nm)".format(w, len(fibers[w]),
+                                                                                             np.median(difference[w]),
+                                                                                             iqr(difference[
+                                                                                                     w]) / 1.349),
+                        )
+        plt.legend(fontsize=7)
+        plt.xlabel("fiberId")
+        plt.ylabel("Wavelength difference ({}-{}) [nm]".format(arm[0], arm[1]))
+        plt.savefig("overlapLines-{:06}-{}{}{}.png".format(visit, arm[0], arm[1], spectrograph))
+        plt.close()
+
+        return Struct()
+
+    @classmethod
+    def _makeArgumentParser(cls) -> ArgumentParser:
```

Comment:
You can remove this method (_makeArgumentParser) from this class because this is a `Task` that is not `CmdLineTask`.

Reply:
I haven't really touched the Overlapping lines regions code and @kiyoyabe is going to make a new jira issue to deal with it so I'll just leave this for now, but thanks.

Reply:
https://pfspipe.ipmu.jp/jira/browse/PIPE2D-1375